packages/@lwc/synthetic-shadow/src/faux-shadow/node.ts

```diff
@@ -300,7 +300,16 @@ defineProperties(Node.prototype, {
             if (isNodeShadowed(this)) {
                 return parentNodeGetterPatched.call(this);
             }
-            return parentNodeGetter.call(this);
+
+            const parentNode = parentNodeGetter.call(this);
+
+            // Handle the case where a top level light DOM element is slotted into a synthetic
+            // shadow slot.
+            if (!isNull(parentNode) && isSlotElement(parentNode) && isNodeShadowed(parentNode)) {
```

Comment:
```suggestion
            if (!isNull(parentNode) && isSyntheticSlotElement(parentNode)) {
```

`isSyntheticSlotElement` does the same thing as `isSlotElement(node) && isNodeShadowed(node)`.